File: Python/cars.py
class Vehicle:
    def __init__(self, type, brand, model, year, color, plate):
        self.type = type
        self.brand = brand
        self.model = model
        self.year = year
        self.color = color
        self.plate = plate
        


def add_vehicle(vehicle_list):
    if len(vehicle_list) >= 100:
        print("La limite du nombre de véhicules a été atteinte.")
        return

    type = input("Type de Véhicule : ")
    brand = input("Marque : ")
    model = input("Modèle : ")
    year = input("Année : ")
    color = input("Couleur : ")
    plate = input("Numéro d'immatriculation : ")

    new_vehicle = Vehicle(type, brand, model, year, color, plate)
    vehicle_list.append(new_vehicle)

    print("Le véhicule a été ajouté avec succès.")


def remove_vehicle(vehicle_list):
    if len(vehicle_list) == 0:
        print("Aucun véhicule enregistré.")
        return

    search_plate = input("Entrez le numéro d'immatriculation du véhicule à supprimer : ")

    found_index = -1
    for i, vehicle in enumerate(vehicle_list):
        if vehicle.plate == search_plate:
            found_index = i
            break

    if found_index == -1:
        print("Aucun véhicule avec ce numéro d'immatriculation trouvé.")
    else:
        del vehicle_list[found_index]
        print("Le véhicule a été supprimé avec succès.")


def display_statistics(vehicle_list):
    if len(vehicle_list) == 0:
        print("Aucun véhicule enregistré.")
        return

    num_cars = 0
    num_motorcycles = 0
    num_buses = 0
    num_trains = 0

    num_red_cars = 0
    num_blue_cars = 0
    num_green_cars = 0
    # Ajouter plus de variables de couleur au besoin

    for vehicle in vehicle_list:
        if vehicle.type == "Voiture":
            num_cars += 1
            if vehicle.type == "Rouge":
                num_red_cars += 1
            elif vehicle.color == "Bleu":
                num_blue_cars += 1
            elif vehicle.color == "Vert":
                num_green_cars += 1
        elif vehicle.type == "Moto":
            num_motorcycles += 1
        elif vehicle.type == "Autobus":
            num_buses += 1
        elif vehicle.type == "Métro":
            num_trains += 1

    print("----- Statistiques -----")
    print("Nombre de voitures :", num_cars)
    print("Nombre de motos :", num_motorcycles)
    print("Nombre d'autobus :", num_buses)
    print("Nombre de métros :", num_trains)
    print("Nombre de voitures rouges :", num_red_cars)
    print("Nombre de voitures bleues :", num_blue_cars)
    print("Nombre de voitures vertes :", num_green_cars)
    print("------------------------")


def display_vehicles(vehicle_list):
    if len(vehicle_list) == 0:
        print("Aucun véhicule enregistré.")
        return

    print("----- Liste des véhicules -----")
    for vehicle in vehicle_list:
        print("Type :", vehicle.type)
        print("Marque :", vehicle.brand)
        print("Modèle :", vehicle.model)
        print("Année :", vehicle.year)
        print("Couleur :", vehicle.color)
        print("Numéro d'immatriculation :", vehicle.plate)
        print("-------------------------------")


def search_vehicle_by_plate(vehicle_list):
    search_plate = input("Entrez le numéro d'immatriculation à rechercher : ")

    print("----- Résultats de la recherche -----")
    for vehicle in vehicle_list:
        if vehicle.plate == search_plate:
            print("Marque :", vehicle.type)
            print("Marque :", vehicle.brand)
            print("Modèle :", vehicle.model)
            print("Année :", vehicle.year)
            print("Couleur :", vehicle.color)
            print("Numéro d'immatriculation :", vehicle.plate)
            print("-----------------------------------")
            return

    print("Aucun véhicule avec ce numéro d'immatriculation trouvé.")

File: C/cars.c
#include <stdio.h>
#include <string.h>
#include "cars.h"

void addVehicle(Vehicle vehicleList[], int* numVehicles) {
    if (*numVehicles == MAX_VEHICLES) {
        printf("La limite de véhicules a été atteinte.\n");
        return;
    }

    Vehicle newVehicle;
    
    printf("Type de Véhicule : ");
    scanf("%49s", newVehicle.type);

    printf("Marque : ");
    scanf("%49s", newVehicle.brand);

    printf("Modèle : ");
    scanf("%49s", newVehicle.model);

    printf("Année : ");
    scanf("%9s", newVehicle.year);

    printf("Couleur : ");
    scanf("%19s", newVehicle.color);

    printf("Numéro d'immatriculation : ");
    scanf("%19s", newVehicle.plate);

    vehicleList[*numVehicles] = newVehicle;
    *numVehicles = *numVehicles + 1;

    printf("Le véhicule a été ajouté avec succès.\n");

}

void removeVehicle(Vehicle vehicleList[], int* numVehicles) {
    char searchPlate[MAX_PLATE_LENGTH];
    printf("Entrez le numéro d'immatriculation du véhicule à supprimer : ");
    scanf("%19s", searchPlate);

    int foundIndex = -1;

    for (int i = 0; i < *numVehicles; i++) {
        if (strcmp(vehicleList[i].plate, searchPlate) == 0) {
            foundIndex = i;
            break;
        }
    }

    if (foundIndex != -1) {
        for (int i = foundIndex; i < *numVehicles - 1; i++) {
            vehicleList[i] = vehicleList[i + 1];
        }
        *numVehicles = *numVehicles - 1 ;
        printf("Le véhicule a été supprimé avec succès.\n");

    } 
    
    else {
        printf("Aucun véhicule avec ce numéro d'immatriculation trouvé.\n");
    }
}

void displayStatistics(Vehicle vehicleList[], int numVehicles) {
    int numCars = 0;
    int numMotorcycles = 0;
    int numBuses = 0;
    int numTrains = 0;
    int numRedCars = 0;
    int numBlueCars = 0;
    int numGreenCars = 0;
    // Add more color variables as needed

    for (int i = 0; i < numVehicles; i++) {
        if (strcmp(vehicleList[i].type, "Voiture") == 0) {
            numCars++;
            if (strcmp(vehicleList[i].color, "Rouge") == 0)
                numRedCars++;
            else if (strcmp(vehicleList[i].color, "Bleu") == 0)
                numBlueCars++;
            else if (strcmp(vehicleList[i].color, "Vert") == 0)
                numGreenCars++;
        } else if (strcmp(vehicleList[i].type, "Moto") == 0) {
            numMotorcycles++;
        } else if (strcmp(vehicleList[i].type, "Autobus") == 0) {
            numBuses++;
        } else if (strcmp(vehicleList[i].type, "Métro") == 0) {
            numTrains++;
        }
    }

    printf("Statistiques de circulation routière :\n");
    printf("Nombre de voitures : %d\n", numCars);
    printf("Nombre de motos : %d\n", numMotorcycles);
    printf("Nombre d'autobus : %d\n", numBuses);
    printf("Nombre de métros : %d\n", numTrains);
    printf("Nombre de voitures rouges : %d\n", numRedCars);
    printf("Nombre de voitures bleues : %d\n", numBlueCars);
    printf("Nombre de voitures vertes : %d\n", numGreenCars);
}

void displayVehicles(Vehicle vehicleList[], int numVehicles) {
    if (numVehicles == 0) {
        printf("Aucun véhicule enregistré.\n");
        return;
    }

    printf("Liste des véhicules enregistrés :\n");
    for (int i = 0; i < numVehicles; i++) {
        printf("Véhicule %d\n", i + 1);
        printf("Type : %s\n", vehicleList[i].type);
        printf("Marque : %s\n", vehicleList[i].brand);
        printf("Modèle : %s\n", vehicleList[i].model);
        printf("Année : %s\n", vehicleList[i].year);
        printf("Couleur : %s\n", vehicleList[i].color);
        printf("Numéro d'immatriculation : %s\n", vehicleList[i].plate);
        printf("--------------------\n");
    }
}

void searchVehicleByPlate(Vehicle vehicleList[], int numVehicles) {
    char searchPlate[MAX_PLATE_LENGTH];
    printf("Entrez le numéro d'immatriculation du véhicule à rechercher : ");
    scanf("%19s", searchPlate);

    int foundIndex = -1;

    for (int i = 0; i < numVehicles; i++) {
        if (strcmp(vehicleList[i].plate, searchPlate) == 0) {
            foundIndex = i;
            break;
        }
    }

    if (foundIndex != -1) {
        printf("Véhicule trouvé :\n");
        printf("Type : %s\n", vehicleList[foundIndex].type);
        printf("Marque : %s\n", vehicleList[foundIndex].brand);
        printf("Modèle : %s\n", vehicleList[foundIndex].model);
        printf("Année : %s\n", vehicleList[foundIndex].year);
        printf("Couleur : %s\n", vehicleList[foundIndex].color);
        printf("Numéro d'immatriculation : %s\n", vehicleList[foundIndex].plate);
    } else {
        printf("Aucun véhicule avec ce numéro d'immatriculation trouvé.\n");
    }
}

void searchVehicleByType(Vehicle vehicleList[], int numVehicles) {
    char searchType[MAX_TYPE_LENGTH];
    printf("Entrez le type de véhicule a rechercher : ");
    scanf("%19s", searchType);

    int foundIndex = -1;

    for (int i = 0; i < numVehicles; i++) {
        if (strcmp(vehicleList[i].type, searchType) == 0) {
            foundIndex = i;
            break;
        }
    }

    if (foundIndex != -1) {
        printf("Véhicule trouvé :\n");
        printf("Type : %s\n", vehicleList[foundIndex].type);
        printf("Marque : %s\n", vehicleList[foundIndex].brand);
        printf("Modèle : %s\n", vehicleList[foundIndex].model);
        printf("Année : %s\n", vehicleList[foundIndex].year);
        printf("Couleur : %s\n", vehicleList[foundIndex].color);
        printf("Numéro d'immatriculation : %s\n", vehicleList[foundIndex].plate);
    } else {
        printf("Aucun véhicule de ce type trouvé.\n");
    }
}

